destination/retry.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/avast/retry-go/v4"
+)
+
+// RetryNetError retries the given operation only if it returns a net.Error.
+// Any other error will be returned immediately.
+func RetryNetError(op retry.RetryableFunc, ctx context.Context, opName string) error {
+	err := retry.Do(
+		op,
+		retry.Attempts(*maxRetries),
+		retry.DelayType(retry.FixedDelay),
```

Comment:
Did you consider exponential backoff?

Reply:
Yes, that's actually was in the initial implementation of `retry.go`, I decided to simplify it for time being. Maybe you are right, we should implement it straight away.

Reply:
Updated with exponential backoff.